zjit/src/cruby_methods.rs

```diff
@@ -275,6 +276,25 @@ fn inline_string_getbyte(fun: &mut hir::Function, block: hir::BlockId, recv: hir
     None
 }
 
+fn inline_string_eq(fun: &mut hir::Function, block: hir::BlockId, recv: hir::InsnId, args: &[hir::InsnId], state: hir::InsnId) -> Option<hir::InsnId> {
+    let &[other] = args else { return None; };
+    if fun.likely_a(recv, types::String, state) && fun.likely_a(other, types::String, state) {
+        let recv = fun.coerce_to(block, recv, types::String, state);
+        let other = fun.coerce_to(block, other, types::String, state);
+        let return_type = types::BoolExact;
+        let elidable = true;
+        let result = fun.push_insn(block, hir::Insn::CCall {
```

Comment:
I think I should make this its own opcode, or at least leave a TODO to do that, so we can const-fold `a == a` type comparisons.